temporal-kotlin/src/main/kotlin/io/temporal/worker/WorkerExt.kt

```diff
@@ -75,6 +80,26 @@ inline fun <reified T : Any> Worker.addWorkflowImplementationFactory(
 }
 
 /**
+ * Configures a factory to use when an instance of a workflow implementation is created.
+ * Please read an original [Worker.registerWorkflowImplementationFactory] method doc because
+ * this method has a limited usage.
+ *
+ * @param T Workflow interface that this factory implements
+ * @param factory factory that when called creates a new instance of the workflow implementation
+ * object.
+ * @param options custom workflow implementation options for a worker
+ * @see Worker.registerWorkflowImplementationFactory
+ */
+inline fun <reified T : Any> Worker.registerWorkflowImplementationFactory(
+  noinline factory: () -> T,
+  options: WorkflowImplementationOptions
+) {
+  registerWorkflowImplementationFactory(T::class.java, factory, options)
+}
```

Comment:
Is it possible to keep `factory` as the last parameter here? That would enable a nicer experience for calling this method.

**`factory` as the first parameter**
```kotlin
worker.registerWorkflowImplementationFactory({ /* code to create workflow instance */ }, options)
```

**`factory` as the last parameter**
```kotlin
worker.registerWorkflowImplementationFactory(options) { /* code to create workflow instance */ }
```